### DISCUSSION TITLE
Enable codeGeneration on vercel?

### DISCUSSION THREAD

**Discussion by CapitaineToinon** [vercel/edge-runtime#459] [Q&A] [answered]
Hey,

I've been trying for a few hours to generate MDX files on edge through a nextjs app on Vercel. I went down quite the rabbit hole to find out why this didn't work as the errors on happened in production and weren't helpful. 

I ended up in Next.js doc telling me that `eval` and `new Function` are disabled on the edge. I understand the security reasons but I also found out through this repo that [there is an option to enable code generation](https://github.com/vercel/edge-runtime/blob/main/packages/vm/src/vm.ts#L9) but couldn't find any documentation anywhere.

Packages such as [@mdx-js/mdx](https://mdxjs.com/packages/mdx/) and [hashicorp/next-mdx-remote](https://github.com/hashicorp/next-mdx-remote) rely on evaluation of JavaScript.

Is there no way to get this working somehow?

**Comment by Kikobeats** [comment, marked as answer]
Hello,

The code generation parameter you're referenced is just for emulation and local development.

The dynamic code execution limits are defined here
https://vercel.com/docs/concepts/functions/edge-functions/limitations#dynamic-code-execution-leads-to-a-runtime-error

It isn't possible to bypass that limits

**Reply by CapitaineToinon** [reply]
I see, that's unfortunate. Thank you for the swift answer!
